### DISCUSSION TITLE
Application performance monitoring

### DISCUSSION THREAD

**Discussion by jharding** [vercel/next.js#14666] [Help]
What's the state of the art when it comes to APM with next.js apps? I went digging around the other day to learn what folks were using and stumbled upon this thread which was a little disheartening: https://github.com/DataDog/dd-trace-js/issues/934.

Is anyone using something they're happy with? I'm curious to know what the Vercel team does for monitoring.

**Comment by patjakubik** [comment]
After spending some time googling, it looks like no-one heard of monitoring Next.js apps. It's been half year since your post, did you figured it out?

**Comment by michrome** [comment]
Hey @jharding and @patryk-smc if your Next.js app runs on Vercel, you can use [Log Drains](https://vercel.com/blog/log-drains). Datadog is one of the [logging integrations in the marketplace](https://vercel.com/integrations?category=logging).

Also: [Log Drains API documentation](https://vercel.com/docs/api#integrations/log-drains).

**Comment by rshiny20** [comment]
Rightfully touted as the 'Mozilla Firefox' of the APM industry, Atatus makes application monitoring easier with simple, actionable insights. You can monitor your entire application in a single interface. And spot errors faster using advanced diagnostics that tell you exactly what went wrong and how to fix it.
